packages/itwin/grouping-mapping-widget/src/widget/components/GroupQueryBuilderContainer.tsx

```diff
@@ -41,21 +51,32 @@ export const GroupQueryBuilderContainer: React.FunctionComponent = () => {
     return () => {
       GroupQueryBuilderApi.removeSelectionListener();
```

Comment:
Adding/removing listeners through this "Api" class is asking for trouble. I think it would be better to remove those wrappers. `GroupAction.tsx` adds listeners directly, so I guess this component also can do it the same way.